### DISCUSSION TITLE
Examples

### DISCUSSION THREAD

**Discussion by pikachu937** [bluenviron/mediamtx#2816] [Questions]
## Describe the feature

Please add examples of working with each protocol and different programs (ffmpeg/gstreamer/obs). The documentation says "Known clients that can publish using RTSP: FFmpeg, GStreamer, OBS Studio", but there is not a single example of playing whep(s) with "FFmpeg, GStreamer, OBS Studio". I would also really like to see real examples of whep(s) working with mobile platforms (android/ios) or at least links to examples without using webview, which in itself requires the use of a browser, which not everyone has updated and, as a result, the video is not can be received.

**Comment by rgl** [comment]
FWIW, I got a couple of examples at https://github.com/rgl/mediamtx-rtsp-server

**Reply by iconoclasthero** [reply]
Where?

**Comment by pikachu937** [comment]
> FWIW, I got a couple of examples at https://github.com/rgl/mediamtx-rtsp-server

what if in the context of using the same whep in mobile platforms without webview?

**Comment by rgl** [comment]
> > FWIW, I got a couple of examples at https://github.com/rgl/mediamtx-rtsp-server
> 
> what if in the context of using the same whep in mobile platforms without webview?

Sorry, but I do not known about mobile. I've only tried with mainstream/evergreen desktop browsers.

**Comment by pikachu937** [comment]
> > > FWIW, I got a couple of examples at https://github.com/rgl/mediamtx-rtsp-server
> > 
> > 
> > what if in the context of using the same whep in mobile platforms without webview?
> 
> Sorry, but I do not known about mobile. I've only tried with mainstream/evergreen desktop browsers.

There are no problems with PC, there are problems when developing an application for mobile platforms. I didn’t find a single implementation of webrtc for a mobile phone that uses whep in its pure form (that’s why I asked for examples), only webview, which is not entirely correct because not all phones are updated and, as a result, the browser in which whep is played cannot even connect to the stream because , that does not know about methods in js or software support for codecs was not enabled at the hardware level in the old version of the system. pure wss would definitely help here (which, by the way, could be used to improve sip operation), but wss was removed from the webrtc implementation. as a result, you have to invent strange “crutches” instead of simply getting the flow

**Comment by ropucyka** [comment]
@pikachu937 

i cannot even `ffplay.exe -i http://127.0.0.1:8889/mystream/whep`
is just not opening and says about web page error.
Firefox v116 just shows black player screen with 0:00 time and mediamtx console says about "not supported codec by client".

And idk where is problem - in obs, in mediamtx, in player? or in this whole clown protocol? And it have problems with connections configuration.

[a comment by an automated account is not shown]
